### PR TITLE
2586 deleting a class with references gives extra do you want to delete anyway msg

### LINKED ISSUE #2586: Deleting a class with references gives extra "Do you want to delete anyway" msg
When you delete a class that has references to it - Pharo helpfully asks you if you want to view the references (which is cool) - if you answer No - you get a superfluous extra question "Do you want to remove anyway?".  This is annoying as if you say yes - you then get the refactoring window that shows you the removal which you can also accept and cancel. Thus you don't need that extra question as you can still cancel.

### PR DESCRIPTION
**Pull request by gcorriga**
Streamline the behaviour of the browser when removing classes, avoiding raising multiple dialogs.

The new behaviour is as follows:

1) if the class has no references, subclasses, or users, perform a refactoring to remove the class
2) if the class has references or users, ask the user what to do - browse the references/users, or do nothing.
3) if the class has subclasses, ask the user what to do - delete keeping subclasses, browse the subclasses, or do nothing.

As part of this change, remove the menu item to remove a class keeping subclasses since its behaviour is now integrate in the standard remove workflow.

Fixes #2586.

### CODE REVIEW

File: src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st [hunk 1]
```diff
@@ -105,12 +87,61 @@ ClyBrowserMorph >> confirmUnusedVariablesInDefiningClass: variables [
...
+	 | hasReferences hasSubclasses hasUsers |
+	 hasReferences := self hasReferencesToClasses: classes.
+	 hasSubclasses := ((classes flatCollect: [ :each | each subclasses ]) 
+		                   copyWithoutAll: classes) notEmpty.
+	 hasUsers := ((classes flatCollect: [ :each | each users ]) 
+		              copyWithoutAll: classes) notEmpty.
+
+	 hasReferences | hasSubclasses | hasUsers ifFalse: [ 
+		 ^ {SycSilentlyRemoveClassStrategy new} ].
+
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st:120]: Cool that you follow the method removal approach. 
Now with your nice changes and tests I think this logic for references analysis can be moved to SycRemoveClassStrategy (and same refactoring for method removal logic).
(some thoughts for possible improvements)

File: src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st [hunk 2]
```diff
@@ -105,12 +87,61 @@ ClyBrowserMorph >> confirmUnusedVariablesInDefiningClass: variables [
...
+{ #category : #'*Calypso-SystemTools-Core' }
+ClyBrowserMorph >> hasReferencesToClasses: classes [
+	
+	| query references |
+	
+	query := (ClyClassReferencesQuery toAny: classes from: self systemScope).
+	references := query execute.	
+	^(references notEmpty and: [
+		((references items collect: [:each | each origin])
+			difference: (classes flatCollect: [ :each | { each . each classSide } ])) notEmpty])
```
Comment (dionisiydk) [inline src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st:99]: I also thought to extract this logic into ClyClassReferencesQuery itself (or special subclass). 
- query ignoreSelfReferences
- ClyExternalClassReferencesQuery

**Review by dionisiydk — APPROVED** [review]
Good stuff but I did not play with live image

**Comment by MarcusDenker** [comment]
somehow the image can not be build:

Error: This package depends on the following classes:
  ClyBrowserMorphTest
You must resolve these dependencies before you will be able to load these definitions: 
  ClyBrowserMorphTest>>#testHasReferencesToClass
  ClyBrowserMorphTest>>#testRequestRemoveClassStrategyForParentClass
  ClyBrowserMorphTest>>#testRequestRemoveClassStrategyForReferencedClass
  ClyBrowserMorphTest>>#testRequestRemoveClassStrategyForUnreferencedClass
  ClyBrowserMorphTest>>#testRequestRemoveClassStrategyForUsedTrait

**Comment by gcorriga** [comment]
It looks like I got the dependency order mixed up. I'll look into fixing that.

**Comment by gcorriga** [comment]
The tests above pass because the test class causing issue seems to be missing.

Cleanest thing to do is to close the PRs and submit again.

**Pull request closed without merging**